src/win.c

```diff
@@ -2374,6 +2453,44 @@ void win_clear_flags(struct managed_win *w, uint64_t flags) {
 	w->flags = w->flags & (~flags);
 }
 
+void win_set_property_stale(struct managed_win *w, xcb_atom_t prop) {
+	const auto bits_per_element = sizeof(*w->stale_props) * 8;
+	if (prop >= w->stale_props_capacity * bits_per_element) {
+		const auto new_size = prop / bits_per_element + 1;
+		w->stale_props = realloc(w->stale_props, new_size * sizeof(*w->stale_props));
+
+		// Clear the content of the newly allocated bytes
+		memset(w->stale_props + w->stale_props_capacity, 0,
+		       new_size - w->stale_props_capacity);
+		w->stale_props_capacity = new_size;
+	}
```

Comment:
Is it okay to loose all previously set property flags, when we have to expand (i.e. re-allocate) our bitmask?

Reply:
No, the memset only clear the newly allocated bytes.

Reply:
Sorry, my bad. `realloc()` does in fact preserve its content. Had that mistaken with another function.

Reply:
Ah, wait. I forgot to `* sizeof(*w->stale_props)` in the memset.